File: JAVA_CalculadoraReflection/src/br/ifes/projetosistemas/calculadora/model/operation/Dividir.java
package br.ifes.projetosistemas.calculadora.model.operation;

public class Dividir implements IOperation{

    public int calculate(int valor1, int valor2){
        if (valor2 == 0)  throw new ArithmeticException("Não é possível dividir por zero.");
        return valor1 / valor2;
    }
}

File: CSharp_Calculadora/CalculadoraConsole/Model/Dividir.cs
namespace CalculadoraConsole.Model
{
    public class Dividir : IOperacao
    {
        public int Calcular(int Valor1, int Valor2)
        {
            if (Valor2 == 0)
            {
                throw new DivideByZeroException("Não é possível dividir por zero!");
            }
            return Valor1 / Valor2;
        }
    }
}
